### PR TITLE
Workaround for TextPainter.computeDistanceToActualBaseline when the text is empty

### PR DESCRIPTION
**Pull request by HansMuller**
This is a workaround for TextPainter.computeDistanceToActualBaseline when the text is empty, #13303

This problem crops up when Flutter tries to baseline-align a TextField before the user has input anything.

### CODE REVIEW

File: packages/flutter/test/painting/text_painter_rtl_test.dart [hunk 1]
```diff
@@ -658,6 +658,21 @@ void main() {
...
+      text: '',
+      style: const TextStyle(fontFamily: 'Ahem', fontSize: 100.0, height: 1.0),
+    );
+    painter.layout();
+    expect(
+      // Returns -1
+      painter.computeDistanceToActualBaseline(TextBaseline.alphabetic), 80.0,
+      skip: skipExpectsWithKnownBugs,
+    );
+  }, skip: skipTestsWithKnownBugs);
```
Comment (Hixie) [inline packages/flutter/test/painting/text_painter_rtl_test.dart]: with your fix this doesn't actually fail, right?

Comment (HansMuller) [reply packages/flutter/test/painting/text_painter_rtl_test.dart]: With my "fix" the painter returns 84 in this case. As far as I understand it, it should be 80.

84 is a big improvement over -1, however I think it's still wrong.

Comment (Hixie) [reply packages/flutter/test/painting/text_painter_rtl_test.dart]: Ah, right.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**